actors/migration/nv9/top.go

```diff
@@ -223,12 +238,14 @@ type actorMigration interface {
 	// Loads an actor's state from an input store and writes new state to an output store.
 	// Returns the new state head CID.
 	migrateState(ctx context.Context, store cbor.IpldStore, input actorMigrationInput) (result *actorMigrationResult, err error)
+	migratedCodeCID() cid.Cid
```

Comment:
It's a bit annoying that this was needed only for the cached case. I suggest instead making it a member of `cachedMigrator`, set in the `migrations` map in the same way and place it's provided to the `nilMigrator`.

Reply:
It's ugly but it's better in that the Migrator maintains the CID for itself so there is only one piece of logic responsible for the CID returned by both the cached and uncached code paths.